### ISSUE TITLE
[nextgen-editor] Relative links break inside editor

### ISSUE THREAD

**Issue by slwa-epowell** [getgrav/grav-premium-issues#192]
When we create a relative link, e.g. ../../alpha/beta, when an editor clicks on the link from inside the Grav NextGen editor it brings up a Error 404. 

Steps to reproduce:

1. Add a relative link to your content using the link button from the NextGen Editor toolbar.
2. Click on the new link. A tooltip will pop up with a hyperlink to the link location.
3. Click on the hyperlink. A new window will open with an Error 404 page.

The error is because the URL does not include the /pages folder, i.e. when it should be <base_url>/admin/pages/alpha/beta/gamma it has instead loaded <base_url>/admin/alpha/beta/gamma

My suggestion would be to have the hyperlink use the same filters as Grav does when rendering the page. 

We are on Grav v1.7.26.1 and NextGen Content Editor v1.1.7.

Thanks very much. My content editors find the NextGen Content Editor much more user-friendly experience than writing Markdown, despite a few bugs.

**Comment by w00fz** [comment]
Hey @slwa-epowell,

This is technically correct and I'm afraid there isn't much that can be done to change the behavior without breaking everything else. 

What happens is that NextGen stores the original URL `../../alpha/beta` to make sure Grav is capable of resolving it whenever it loads the page on frontend. However, the clickable URL that you see in the popup, is only a 1:1 representation of the original URL from the NextGen perspective. 
This is technically correct to the eyes of the browser because while for Grav `beta` is a route (folder) that can be looked back 1 level with `../`, to the browser `beta` is a "file" path within the `alpha` folder. So when you do `../../alpha/beta` from `<base_url>/admin/pages/alpha/beta` the browsers translates to `<base_url>/admin/alpha/beta`. If the URL ended with a forward slash (`<base_url>/admin/pages/alpha/beta/`) then the browser would know beta is a folder.

Long story short is that changing this behavior would break the actual expected behavior from Grav URI point of view and would really only fix a cosmetic issue. It is also not possible for me to apply the same filter Grav uses because from that point on it would be permanently converted to a fixed path and the original dot-dot notation would be lost.

If it is critical for you that the popup links are reachable at all times, my advice is to instead use the Page Picker to link pages and not rely on relative links. This can be done via the Page Picker icon:
<img width="88" alt="CleanShot 2022-01-10 at 14 36 11@2x" src="https://user-images.githubusercontent.com/11734/148851337-d178954a-1044-4bc6-9879-7db5e010950a.png">

**Comment by slwa-epowell** [comment]
Thanks for looking into this! My vision was that the popup tooltip would apply the filter only on the 'href' portion of the link, but obviously I don't really know how this part of it works.

It's not critical, but this issue was raised by one of our content editors so it's definitely something people notice on their own while editing content. Our Grav site contains almost entirely imported content with relative links, so while I can encourage our content editors to use the Page Picker, it won't help for the majority of pages.

**Comment by slwa-epowell** [comment]
One more update: Even using PagePicker won't solve our problem, as the PagePicker generates an absolute URL that starts with a /, e.g. /alpha/beta. Our Grav installation is inside a subfolder, so those links are also broken when you click on them from the tooltip. 

My content editor was using the tooltip to test that the links work, so they are pretty disappointed that there isn't any way to make them work from inside the editing interface.
